Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -0,0 +1,279 @@
+                string stringToSign = GetStringToSign();
+                string signature = GetSignature();
+                string authorizationHeader = GetAuthorizationHeader();
+
+                return new AwsSignedSubjetToken(verificationRequestUrl.ToString(), StVerificationHttpMethod, "",
+                    canonicalHeaders
+                        .Select(pair => new AwsSignedSubjectTokenHeader(pair.Key, pair.Value))
+                        .Concat(Enumerable.Repeat(new AwsSignedSubjectTokenHeader("Authorization", authorizationHeader), 1))
+                        .ToArray());
+
+                string GetHashedCanonicalRequest()
+                {
+                    // Step 1
+                    // https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+
+                    // First, the HTTP Method.
+                    // See point 1 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    StringBuilder canonicalRequest = new StringBuilder(StVerificationHttpMethod).Append(newLine);
+
+                    // Now the URI's absolute path.
+                    // See point 2 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetPathForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the query string.
+                    // See point 3 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetQueryStringForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the canonical headers.
+                    // We just built them oursevels, so we know header names are lower case and that values
+                    // that could have leading or trailing spaces have been trimmed.
+                    // See point 4 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    foreach (var headerPair in sortedHeaders)
+                    {
+                        canonicalRequest.Append(headerPair.Key);
+                        canonicalRequest.Append(":");
+                        canonicalRequest.Append(headerPair.Value);
+                        canonicalRequest.Append(newLine);
+                    }
+
+                    // Now the list of headers that will be signed.
+                    // See point 5 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(signedHeaders).Append(newLine);
+
+                    // Now the hashed request payload which is empty in our case.
+                    // See point 6 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(GetHexEncodedSha256Hash(Encoding.UTF8.GetBytes("")));
```

Comment:
Presumably we could just hard-code the value here, given that it'll always be the same? I don't mind much either way.

Reply:
(In fact, there's an example hashed empty payload in the docs!)